### PR TITLE
Add subfolders, default trail & content changes

### LINKED ISSUE #11: General feedback on .NET Core site
@shanselman identified several things that should make the .NET Core site much better:
- Introduce better URLs by getting rid of the *.html extensions.
- Fix the Mono explanation on Linux getting started instructions.
- Have a default starting "trail" on the Getting Started page.

I will be making a PR soon with these changes. 

/cc @richlander @akoeplinger @martinwoodward

### PR DESCRIPTION
**Pull request by blackdwarf**
This change adds subfolders to the .NET Core site. This has an effect
on the URLs, which are now extension-less. Also added Linux as the
default trail on the "Getting started" page, so it will load by
default. Changed the text on why Mono is needed for Linux.

Fix #11

/cc @richlander @shanselman @martinwoodward

### CODE REVIEW

File: _includes/core/linux_gs.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
           <p><span class="prompt unix-prompt"></span>sudo apt-get install libunwind8 libssl-dev unzip </p>
         </div>
       </div>
-      <p>Secondly, we need to install Mono (a temporary solution for now).</p>
+      <p>Secondly, we need to install <a href="http://www.mono-project.com">Mono</a> to be able to work with NuGet packages in later steps (a temporary solution for now).</p>
```
Comment (richlander) [inline _includes/core/linux_gs.html]: We can probably be more specific than "work with NuGet packages". It's needed to support the NuGet client, which isn't yet supported by .NET Core.

File: _includes/core/linux_gs.html [hunk 2]
```diff
@@ -20,7 +20,7 @@
           <p><span class="prompt unix-prompt"></span>sudo apt-get install mono-complete </p>
         </div>
       </div>
-      <p>Finally, import certificates for NuGet packages.</p>
+      <p>Finally, import needed certificates for NuGet packages.</p>
```
Comment (richlander) [inline _includes/core/linux_gs.html]: "required"?

**Comment by richlander** [comment]
LGTM, modulo comments.

**Comment by thesmithee** [comment]
Fixed the comment feedback. Also fixed the hero links on the main page.

**Comment by martinwoodward** [comment]
:+1: LGTM

**Pull request merged**